libraries/Wire/src/Wire.cpp

```diff
@@ -412,15 +412,15 @@ void TwoWire::beginTransmission(uint16_t address)
         return;
     }
 #if !CONFIG_DISABLE_HAL_LOCKS
-    if(nonStop && nonStopTask == xTaskGetCurrentTaskHandle()){
```

Comment:
why discard this case? It has actually helped quite a bit, since we require strict use of the Wire API